From 32b77c5078ae73a2cd666ea6ec7f91d95c2c3e83 Mon Sep 17 00:00:00 2001
From: Roy Ivy III <rivy.dev@gmail.com>
Date: Tue, 7 Jun 2016 13:00:26 -0500
Subject: Fix bug RT#114999

* fixes [RT#114999](https://rt.cpan.org/Ticket/Display.html?id=114999)
* 'gnu_compat' mode single character options with optional arguments and default values
  now return correct values when used with no argument from the CLI

Origin: backport, https://github.com/sciurius/perl-Getopt-Long/commit/5d9947fb445327c7299d8beb009d609bc70066c0
Bug: https://rt.cpan.org/Ticket/Display.html?id=114999
Bug-Debian: https://bugs.debian.org/855532
Patch-Name: fixes/getopt-long-1.diff
---
 cpan/Getopt-Long/lib/Getopt/Long.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cpan/Getopt-Long/lib/Getopt/Long.pm b/cpan/Getopt-Long/lib/Getopt/Long.pm
index fdc96bd20..631912bf9 100644
--- a/cpan/Getopt-Long/lib/Getopt/Long.pm
+++ b/cpan/Getopt-Long/lib/Getopt/Long.pm
@@ -1112,7 +1112,7 @@ sub FindOption ($$$$$) {
     if ( $gnu_compat ) {
 	my $optargtype = 0; # 0 = none, 1 = empty, 2 = nonempty
 	$optargtype = ( !defined($optarg) ? 0 : ( (length($optarg) == 0) ? 1 : 2 ) );
-	return (1, $opt, $ctl, undef)
+    return (1, $opt, $ctl, defined($ctl->[CTL_DEFAULT]) ? $ctl->[CTL_DEFAULT] : undef)
 	  if (($optargtype == 0) && !$mand);
 	return (1, $opt, $ctl, $type eq 's' ? '' : 0)
 	  if $optargtype == 1;  # --foo=  -> return nothing
